src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -47,32 +47,67 @@ public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, stri
+                        return null;
+                    }
+                    // The Tool property is typed as Microsoft.VisualStudio.VCProjectEngine.VCCLCompilerTool, and
+                    // we could use it to fetch most of the properties we need, rather than fetching them via the
+                    // IVCRulePropertyStorage interface. However, not all of the properties are exposed directly by
+                    // VCCLCompilerTool (e.g. LanguageStandard). Also, quite a few of the Tool properties are exposed
+                    // as enums, so we'd need to change our code to handle them.
+                    vcFileSettings = vcFileConfig.Tool as IVCRulePropertyStorage;
+                }
+                else
+                {
+                    // header file load file setting from the project
+                    var tools = vcConfig.Tools as IVCCollection;
+                    var toolItem = tools.Item("VCCLCompilerTool");
+                    if (toolItem == null)
+                    {
+                        logger.WriteLine($"Custom built files are not supported. File: '{absoluteFilePath}'");
+                        return null;
+                    }
+                    vcFileSettings = toolItem as IVCRulePropertyStorage;
                 }
-                // The Tool property is typed as Microsoft.VisualStudio.VCProjectEngine.VCCLCompilerTool, and
-                // we could use it to fetch most of the properties we need, rather than fetching them via the
-                // IVCRulePropertyStorage interface. However, not all of the properties are exposed directly by
-                // VCCLCompilerTool (e.g. LanguageStandard). Also, quite a few of the Tool properties are exposed
-                // as enums, so we'd need to change our code to handle them.
-                var vcFileSettings = vcFileConfig.Tool as IVCRulePropertyStorage;
 
                 // Fetch properties that can't be set at file level from the configuration object
                 var includeDirs = vcConfig.GetEvaluatedPropertyValue("IncludePath");
                 var platformToolset = vcConfig.GetEvaluatedPropertyValue("PlatformToolset");
                 var vcToolsVersion = vcConfig.GetEvaluatedPropertyValue("VCToolsVersion");
                 var compilerVersion = GetCompilerVersion(platformToolset, vcToolsVersion);
 
+                // Fetch properties that can be set differently for header files
+                string compileAs = vcFileSettings.GetEvaluatedPropertyValue("CompileAs");
+                string forcedIncludeFiles = vcFileSettings.GetEvaluatedPropertyValue("ForcedIncludeFiles");
+                string precompiledHeader = vcFileSettings.GetEvaluatedPropertyValue("PrecompiledHeader");
+                string precompiledHeaderFile = vcFileSettings.GetEvaluatedPropertyValue("PrecompiledHeaderFile");
+                if (itemType == "ClInclude")
```

Comment:
I believe there can be some refactoring done here that can simplify the method.

At the moment, the method is getting difficult to read and follow what is happening: we already checked itemType above, to get the vcFileSettings. We then use vcFileSettings to calculate some properties. Then we check itemType again, and override the vcFileSettings properties that we just calculated.

Can you please take a look at the method and try to restructure it, so that itemType check and vcFileSettings properties retrieval is simplified?
At the very least, they can be grouped together, without the vcConfig properties block in the middle.